src/elo/util/dbInterface.ts

```diff
@@ -167,11 +180,25 @@ export const writeEloSnapshots = async (
   snapshots: EloSnapshotIM[],
   context: GraphQLContext
 ) => {
-  return context.prisma.eloSnapshot.createMany({
-    data: snapshots,
-  });
+  const upserts = snapshots.map(snapshot =>
+    context.prisma.eloSnapshot.upsert({
+      where: {
+        playerId_matchId_versionId: {
+          playerId: snapshot.playerId,
+          matchId: snapshot.matchId,
+          versionId: snapshot.versionId,
+        },
+      },
+      // do not update existing snapshot
+      update: { },
```

Comment:
```suggestion
      // update existing snapshot if it exists
      update: snapshot,
```

Because this is intended to potentially be a recalculation, it actually _should_ update an existing snapshot. Good catch on the incompleteness of the existing behavior, but I think this should be updated to also edit existing entries.

Reply:
That makes sense.

I have noticed that recalculation also adds on to existing data. To reproduce this in the repl:
> await eloPipeline(...)
> await pullComputedElo(single user)
Wins: 3
Loss: 1
> await eloPipeline(...)
> await pullComputedElo(single user)
Wins: 6
Loss: 2

If this makes sense

Reply:
That does make sense, but is a separate issue to the one you're addressing here, since `EloSnapshot` and `ComputedElo` are distinct tables.

Reply:
Not sure how this can be addressed, if it needs to. I left it for now.